R/read_osw.R

```diff
@@ -164,11 +165,19 @@ fetchPrecursorsInfo <- function(filename, runType = "DIA_Proteomics", selectIDs
   }
 
   # Run query to get peptides, their coordinates and scores.
-  precursorsInfo <- tryCatch(expr = { output <- DBI::dbSendQuery(con, statement = query)
-                     if(all && runType != "DIA_Metabolomics") {DBI::dbBind(output, list("CONTEXT"=context, "FDR"=maxPeptideFdr))}
-                                        DBI::dbFetch(output)},
-                             finally = {DBI::dbClearResult(output)
-                             DBI::dbDisconnect(con)})
+  if (runType=="DIA_IPF"){
+    precursorsInfo <- tryCatch(expr = { output <- DBI::dbSendQuery(con, statement = query)
+    if(all && runType != "DIA_Metabolomics") {DBI::dbBind(output, list("CONTEXT"=context, "FDR"=maxPeptideFdr, "USE_IDENTIFYING"=useIdentifying))}
+    DBI::dbFetch(output)},
+    finally = {DBI::dbClearResult(output)
+      DBI::dbDisconnect(con)})
+  } else {
+    precursorsInfo <- tryCatch(expr = { output <- DBI::dbSendQuery(con, statement = query)
+                       if(all && runType != "DIA_Metabolomics") {DBI::dbBind(output, list("CONTEXT"=context, "FDR"=maxPeptideFdr))}
```

Comment:
I think we can just add another if statement here to accommodate `DIA_IPF` instead of repeating the lines?